drivers/flash/flash_stm32_qspi.c

```diff
@@ -881,13 +1195,27 @@ static const struct flash_stm32_qspi_config flash_stm32_qspi_cfg = {
 
 static struct flash_stm32_qspi_data flash_stm32_qspi_dev_data = {
 	.hqspi = {
+#ifndef STM32_OSPI_MIMIC_QSPI
 		.Instance = (QUADSPI_TypeDef *)DT_REG_ADDR(STM32_QSPI_NODE),
+#else
+		.Instance = (OCTOSPI_TypeDef *)DT_REG_ADDR(STM32_QSPI_NODE),
+#endif
 		.Init = {
 			.FifoThreshold = STM32_QSPI_FIFO_THRESHOLD,
+#ifndef STM32_OSPI_MIMIC_QSPI
 			.SampleShifting = QSPI_SAMPLE_SHIFTING_NONE,
 			.ChipSelectHighTime = QSPI_CS_HIGH_TIME_1_CYCLE,
 			.ClockMode = QSPI_CLOCK_MODE_0,
-			},
+#else
+#if DT_PROP(STM32_QSPI_NODE, chip_select_boundary) > 0
+#error OSPI ChipSelectBoundary needs to be zero
+#endif
+			.SampleShifting = HAL_OSPI_SAMPLE_SHIFTING_NONE,
+			.ChipSelectHighTime = 1,
+			.ClockMode = HAL_OSPI_CLOCK_MODE_0,
+			.ClockPrescaler = 1,
```

Comment:
There isn't a define for the prescaler?

Reply:
No, there isn't. Here is the description from the HAL libraries, where it usually also shows which define to use.
```
  uint32_t ClockPrescaler;            /*!< It specifies the prescaler factor used for generating
                                           the external clock based on the AHB clock.
                                           This parameter can be a value between 1 and 256 */
```